backend/pkg/api/handlers/auth.go

```diff
@@ -99,6 +100,58 @@ Best regards,
 	return nil
 }
 
+// TODO move to service?
+func (h *HandlerService) sendResetEmail(ctx context.Context, userId uint64, email string) error {
+	// 0. check if email resets are allowed
+	// (can be forbidden by admin (not yet in v2))
+	passwordResetAllowed, err := h.dai.IsPasswordResetAllowed(ctx, userId)
+	if err != nil {
+		return err
+	}
+	if !passwordResetAllowed {
+		return newForbiddenErr("password reset not allowed")
+	}
+
+	// 1. check last confirmation time to enforce ratelimit
+	lastTs, err := h.dai.GetPasswordResetTime(ctx, userId)
+	if err != nil {
+		return errors.New("error getting confirmation-ts")
+	}
+	if lastTs.Add(authResetEmailRateLimit).After(time.Now()) {
+		return errors.New("rate limit reached, try again later")
```

Comment:
This would return 500 to the user, something like 429 would be more appropriate.